lib/Backend/amd64/LowererMDArch.h

```diff
@@ -29,7 +29,61 @@ class LowererMDArch
     int                 helperCallArgsCount;
     IR::Opnd *          helperCallArgs[MaxArgumentsToHelper];
 
-
+#ifndef _WIN32
+    class XPlatRegArgList
+    {
+        bool isRecording;
+        int lastPosition;
+    public:
+        XPlatRegArgList():isRecording(false) { Reset(); }
+        inline void Reset() { floatCount = 0; intCount = 0; lastPosition = 0;}
+
+        inline void StartRecording() { Assert(!isRecording); Reset(); isRecording = true; }
+        
+        void StopRecording()
+        {
+            Assert(isRecording);
+            isRecording = false;
+            for (int i = lastPosition - GetArgsCount(); i >= 0; i--) args[i] = TyMachPtr;
```

Comment:
There are some assumptions hidden in this code. Looks this code assumes the recording pattern must be we always record every register in the back range but none in the front range if any is missing recording. That's why you clear range [0, _start_). Is it right? Is there a way to ensure this assumption holds? At least can you document it?